### ISSUE TITLE
Error using miles for percision in Geo Location Context

### ISSUE THREAD

**Issue by KellyBennett** [elastic/elasticsearch#24807]
**Elasticsearch version**:
5.1.2

**Description of the problem including expected versus actual behavior**:
Here is the search I want to run:

```
POST stores_search/_search?
{
  "suggest": {
    "store-suggest": {
      "prefix": "Combs",
      "completion": {
        "field": "suggest",
        "fuzzy": {
          "fuzziness": "AUTO",
          "transpositions": true
        },
        "size": 100,
        "contexts": {
          "location": [
            {
              "lat": 47.6062,
              "lon": -122.3321,
              "precision": "100m"
            }
          ]
        }
      }
    }
  }
}
```

I need to provide my precision in distance. The docs state this is possible:
```
The precision of the geohash to encode the query geo point. This can be specified as a distance value (5m, 10km etc.), or as a raw geohash precision (1..12). Defaults to index time precision level.
```

But instead of getting stores within that geo location, I get this error response:
```
{
  "error": {
    "root_cause": [
      {
        "type": "parsing_exception",
        "reason": "[geo] failed to parse field [precision]",
        "line": 1,
        "col": 57
      }
    ],
    "type": "search_phase_execution_exception",
    "reason": "all shards failed",
    "phase": "query_fetch",
    "grouped": true,
    "failed_shards": [
      {
        "shard": 0,
        "index": "stores_search",
        "node": "Se4Iz_2cS4OGdz33kndgGg",
        "reason": {
          "type": "parsing_exception",
          "reason": "[geo] failed to parse field [precision]",
          "line": 1,
          "col": 57,
          "caused_by": {
            "type": "number_format_exception",
            "reason": "For input string: \"1000mi\""
          }
        }
      }
    ],
    "caused_by": {
      "type": "parsing_exception",
      "reason": "[geo] failed to parse field [precision]",
      "line": 1,
      "col": 57,
      "caused_by": {
        "type": "number_format_exception",
        "reason": "For input string: \"1000mi\""
      }
    }
  },
  "status": 400
}
```

If i change precision to `1` (or any number in (1..12)), my search completes with no errors.

**Steps to reproduce**:

 1. Create a mapping as an example try:
```
PUT place
{
    "mappings": {
        "shops" : {
            "properties" : {
                "suggest" : {
                    "type" : "completion",
                    "contexts": [
                        { 
                            "name": "place_type",
                            "type": "category",
                            "path": "cat"
                        },
                        { 
                            "name": "location",
                            "type": "geo",
                            "precision": 4
                        }
                    ]
                }
            }
        }
    }
}
```
 2. Put some documents in the new index, as an example:
```
PUT place/shops/1
{
    "suggest": {
        "input": "timmy's",
        "contexts": {
            "location": [
                {
                    "lat": 43.6624803,
                    "lon": -79.3863353
                },
                {
                    "lat": 43.6624718,
                    "lon": -79.3873227
                }
            ]
        }
    }
}
```
 3. Do a lil' query:
```
POST place/_suggest?pretty
{
    "suggest" : {
        "prefix" : "tim",
        "completion" : {
            "field" : "suggest",
            "size": 10,
            "contexts": {
                "location": [ 
                    {
                        "lat": 43.6624803,
                        "lon": -79.3863353,
                        "precision": "2m"
                    },
                    {
                        "context": {
                            "lat": 43.6624803,
                            "lon": -79.3863353
                        },
                        "boost": 2
                    }
                 ]
            }
        }
    }
}
```

result:
```
{
  "error": {
    "root_cause": [
      {
        "type": "parsing_exception",
        "reason": "[geo] failed to parse field [precision]",
        "line": 1,
        "col": 62
      }
    ],
    "type": "search_phase_execution_exception",
    "reason": "all shards failed",
    "phase": "query",
    "grouped": true,
    "failed_shards": [
      {
        "shard": 0,
        "index": "place",
        "node": "Se4Iz_2cS4OGdz33kndgGg",
        "reason": {
          "type": "parsing_exception",
          "reason": "[geo] failed to parse field [precision]",
          "line": 1,
          "col": 62,
          "caused_by": {
            "type": "number_format_exception",
            "reason": "For input string: \"2m\""
          }
        }
      }
    ],
    "caused_by": {
      "type": "parsing_exception",
      "reason": "[geo] failed to parse field [precision]",
      "line": 1,
      "col": 62,
      "caused_by": {
        "type": "number_format_exception",
        "reason": "For input string: \"2m\""
      }
    }
  },
  "status": 400
}
```

**Provide logs (if relevant)**:

**Comment by dbgrey44** [comment]
Any update on this? I am also facing the same issue.
I am also trying to query in a similar fashion as @KellyBennett. Can anyone point me in the correct direction?

**Comment by wgottschalk** [comment]
I'm also noticing this bug on 5.5 as well.

**Comment by vandanachadha** [comment]
I am using ES version 6,2.0, but still getting the same parsing error for "precision" if value is "5m" etc. Works ok if integer.

**Comment by imotov** [comment]
@vandanachadha it was fixed by #29273 As you can see from labels on the issue it was fixed only in 6.3.0 and above. So, it is expected to get the same parsing error in 6.2.0, which was released 2 months before the issue was fixed.
